src/server/node_manager.c

```diff
@@ -4947,6 +4947,7 @@ is_request(int stream, int version)
 			}
 			if (made_new_vnodes || cr_node) {
 				save_nodes_db(1, pmom); /* update the node database */
+				propagate_licenses_to_vnodes(pmom);
```

Comment:
This must be a duplicate call. The same is invoked in line 4525

Reply:
That call will not be invoked incase of update from hook so I have invoked the call here

Reply:
Okay. Is it possible to remove the first invocation then?

Reply:
No ! ,  that will be used in case of updating vnodes which are not created via hook

Reply:
I agree. But we are not returning from function anywhere in between. So it should hit your newly introduced call as well.

Reply:
Yes but we are inside different cases and condition for both the calls

Reply:
got it. Can you move the first call alongside with save_nodes_db within the same made_new_vnodes and cr_node check then?

Reply:
yes made the change!